test.js

```diff
@@ -213,3 +229,49 @@ test.serial('if once option is true and silent is false, hook only prints the fi
 	t.deepEqual(hookReturnValue, false);
 	t.deepEqual(log, [['foo'], ['bar']]);
 });
+
+test('promise resolves when stdout & stderr are hooked and released via promise', async t => {
+	t.plan(3);
+	const promise = m(str => {
+		if (str === 'foo' || str === 'bar') {
```

Comment:
This test is too loose. I realize it would be caught by `t.plan` if one of the stream were not written to, but would be better to gather the output in an array for each call here and do a deepEqual assertion.